File: Module1/CSharpP2/HW/StringsText/ParseURL/ParseURL.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ParseURL
{
    class ParseURL
    {
        static void Main()
        {
            string url = "http://telerikacademy.com/Courses/Courses/Details/212";
            Console.WriteLine("URL = {0}", url);
            string protocol = url.Substring(0, url.IndexOf("://"));
            Console.WriteLine("[protocol] = {0}", protocol);
            string server = url.Substring(url.IndexOf("://") + 3, url.IndexOf('/', url.IndexOf("://") + 3)  - url.IndexOf("://") - 3);
            Console.WriteLine("[server] = {0}", server);
            string resource = url.Substring(url.IndexOf('/', url.IndexOf("://") + 3));
            Console.WriteLine("[resourece] = {0}", resource);
        }
    }
}

File: Module1/JS-F/HW/Strings/parseURL.js
/*
 Write a script that parses an URL address given in the format:
 [protocol]://[server]/[resource] and extracts from 
 it the [protocol], [server] and [resource] elements.
 Return the elements in a JSON object.*/

String.prototype.parseURL = function () {
    var string = String(this),
        url = {
            protocol: undefined,
            server: undefined,
            resource: undefined
        },
        regExUrlParse = /(\w{3,10}):\/\/([A-z.]+)\/(.+)/g,
        parsedArr = regExUrlParse.exec(string);

    url.protocol = parsedArr[1];
    url.server = parsedArr[2];
    url.resource = parsedArr[3];

    return url;
};

var urlAsStr = 'http:\/\/telerikacademy.com\/Courses\/Courses\/Details\/239';

console.log(urlAsStr.parseURL());